SO-3404: extension FULL RF2 export bug

cmark opened the pull request.

cmark committed:
SO-3404: add getCodeSystemURI() to CodeSystemVersionEntry

cmark committed:
SO-3404: use `String.join` instead of `String.format` in Rf2Exporter

cmark committed:
SO-3404: fix incorrect filtering of candidateVersions

In the previous version of the RF2 exporter, the logic included version
only from the current branch path segment of the extension and dropped
all version from the old branch segments.

cmark committed:
SO-3404: use path for indexing versions instead of name

cmark committed:
SO-3404: fix deserialization issue of CodeSystemVersionEntry

`getCodeSystemURI` is not part of the JSON format of
CodeSystemVersionEntry.

cmark committed:
SO-3404: make sure `created` and `revised` fields are not deserialized..

...when reading revisions (they are read-only/serialized only
properties)

cmark committed:
SO-3404: add basic test to verify FULL/SNAPSHOT content after...

...initial test RF2 import

cmark committed:
SO-3404: remove deprecated ApiError.Builder.of() method

cmark committed:
SO-3404: support to convert SearchIndexResourceRequest to raw search...

...request

cmark committed:
SO-3404: add very basic partial loading search API

nagyo commented on snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedPartialLoadingApiTest.java:
```diff
@@ -0,0 +1,63 @@
+						.execute(context);
+			}
+		)).execute(ApplicationContext.getServiceForClass(Environment.class));
+		System.err.println(hits);
```
Accidentally committed syserr 👀

nagyo committed:
[snomed] Add tests to verify DELTA, SNAPSHOT and FULL export content

cmark committed:
Merge branch '7.x' into issue/SO-3404-extension-full-export-bug

cmark committed:
[index] add back created/revised fields to Revision.Builder

cmark committed:
[rf2] fix incorrect RF2 export when selecting valid descriptions...

...for language code specific refset file

In certain cases the language reference set member revision might not
have a corresponding description revision in the currently exported
branch segment, thus certain RF2 Full rows might not be present in the
resulting export archive.
Use the end range path (the current version basically) to find
descriptions for the current batch of members.

nagyo reviewed (approved):
LGTM!

cmark committed:
SO-3404: remove syserr from SnomedPartialLoadingApiTest
